### PR TITLE
Feature/49060 group agenda items with sections

### PR DESCRIPTION
**Pull request by jjabari-op**
https://community.openproject.org/work_packages/49060

### CODE REVIEW

File: frontend/src/stimulus/controllers/dynamic/meeting-section-form.controller.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  }
+
+  focusInput():void {
+    const titleInput = this.element.querySelector('input[name="meeting_section[title]"]');
+
+    setTimeout(() => {
+      this.element.scrollIntoView({ block: 'center' });
+        (titleInput as HTMLInputElement).focus();
+        this.setCursorAtEnd(titleInput as HTMLInputElement);
+    }, 100);
```
Comment (oliverguenther) [inline frontend/src/stimulus/controllers/dynamic/meeting-section-form.controller.ts]: As the titleInput is already present, why did you add such a high delay? Do we need it at all?

File: modules/meeting/app/contracts/meeting_sections/update_contract.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+module MeetingSections
+  class UpdateContract < BaseContract
+    validate :user_allowed_to_edit
+
+    # attribute :lock_version do
+    #   if model.lock_version.nil? || model.lock_version_changed?
+    #     errors.add :base, :error_conflict
+    #   end
+    # end
```
Comment (oliverguenther) [inline modules/meeting/app/contracts/meeting_sections/update_contract.rb]: Remove, as lock_version not present on sections, and we arguably don't need it

File: modules/meeting/app/controllers/concerns/meetings/agenda_component_streams.rb [hunk 3]
```diff
@@ -86,42 +86,89 @@ def update_sidebar_participants_form_component_via_turbo_stream(meeting: @meetin
           component: Meetings::Sidebar::ParticipantsFormComponent.new(
             meeting:
           ),
-          status: :bad_request
+          status: :bad_request # TODO: why bad_request?
```
Comment (oliverguenther) [inline modules/meeting/app/controllers/concerns/meetings/agenda_component_streams.rb:90]: We want to ensure a non-ok HTTP response so that the modal doesn't auto-close after the form submission. Do you have a specific other status in mind?

File: modules/meeting/app/models/meeting.rb [hunk 4]
```diff
@@ -111,18 +112,18 @@ class Meeting < ApplicationRecord
...
+  #  "work_package_id"=>[nil, 48211],
+  #  "hours"=>[nil, 1.0],
+  #  "comments"=>[nil, "Alex"],
+  #  "activity_id"=>[nil, 8],
+  #  "spent_on"=>[nil, Mon, 31 Jul 2023],
+  #  "tyear"=>[nil, 2023],
+  #  "tmonth"=>[nil, 7],
+  #  "tweek"=>[nil, 31],
+  #  "costs"=>[nil, 0.0],
+  #  "logged_by_id"=>[nil, 9]}
```
Comment (oliverguenther) [inline modules/meeting/app/models/meeting.rb]: wut?

**Review by oliverguenther — COMMENTED** [review]
Except for specs and general cleanup, looking very good already. The way you solved the first section feels very native. Let's try to get this done before we switch all our efforts to open points 👍

File: frontend/src/stimulus/controllers/dynamic/meeting-section-form.controller.ts (hunk 1 above)
Comment (jjabari-op) [reply frontend/src/stimulus/controllers/dynamic/meeting-section-form.controller.ts]: True, we don't need it - I took the implementation from the `agenda-item-form.controller` where it was required at some earlier point during implementation. It can be removed there as well as far as I tested it. Will do so

File: modules/meeting/app/controllers/concerns/meetings/agenda_component_streams.rb (hunk 3 above)
Comment (jjabari-op) [reply modules/meeting/app/controllers/concerns/meetings/agenda_component_streams.rb:90]: This would make sense to me if there's a form which should show from validations - but in this case I don't see that. Additionally, having a static "bad_request" - even for successful submissions - seems like a misuse of the status.

However, I just saw that this method (`update_sidebar_participants_form_component_via_turbo_stream`) is not used at all. Only `update_sidebar_participants_component_via_turbo_stream` is called within the controller.

I would therefore remove `update_sidebar_participants_form_component_via_turbo_stream`. @oliverguenther Or do I miss something here?

File: modules/meeting/app/models/meeting.rb (hunk 4 above)
Comment (jjabari-op) [reply modules/meeting/app/models/meeting.rb]: This was coming from this commit: https://github.com/opf/openproject/commit/5dccda332f296bebbc6b5254f13f5a976f9c6156#diff-d3ed1fc17cdedd062d9753488fd4ed4a8d657ecce4c646246f33fb0a93c08db0

My rubocop extension seems to have corrected the indentation of this comment - thus it's part of the diff in this PR.

I will remove the whole comment as I guess it's not supposed to stay there.

File: modules/meeting/app/controllers/concerns/meetings/agenda_component_streams.rb (hunk 3 above)
Comment (oliverguenther) [reply modules/meeting/app/controllers/concerns/meetings/agenda_component_streams.rb:90]: Yes I think it can be removed, it's a remnant from before the time we rendered the dialog/form separately

**Review by oliverguenther — APPROVED** [review]
Good to merge once the first implicit section is removed again as the spec correctly remarks 👍
